src/strategy.js

```diff
@@ -0,0 +1,26 @@
+import { round, sum, multiply } from './math_bundle.js';
+
+export function scalePrioritisation(matrix, population, vaccinesAvalable) {
+  const wholePopulation = sum(population);
+  const targetCoverage = vaccinesAvalable / wholePopulation;
+  const currentCoverage = sum(
+    multiply(
+      matrix.map(ageGroup => ageGroup[ageGroup.length - 1]),
+      population
+    )
+  ) / wholePopulation;
+
+  //create copy of the matrix
+  let scaled = matrix.map(row => row.slice());
+
+  if (currentCoverage > targetCoverage) {
+    const ratio = targetCoverage / currentCoverage;
+    for (let i = 0; i < scaled.length; i++) {
+      for (let j = 0; j < scaled[i].length; j++) {
+        scaled[i][j] *= ratio;
```

Comment:
I think this line might invalidate one of the caveats I mentioned.
The age-groups (columns) in the row of the matrix we need to adjust (the first one where currentCoverage>targetCoverage) can't all be rescaled. It should only be those age groups that were increased in that prioritisation step.

I've mocked up an example in R (sorry - not confident enough to do it in JS):
```

matrix <- matrix(c(0, 0, 0.8, 0, 0.8, 0.8), nrow = 2, byrow = TRUE)
population <- c(100, 100 , 100)
vaccinesAvailable <- 100

wholePopulation = sum(population)
targetCoverage = vaccinesAvailable / wholePopulation

# Estimate coverage for each step
currentCoverage = c()
for(i in 1:nrow(matrix)){
  currentCoverage[i] <- sum(matrix[i,] * population) / wholePopulation
}
# Identify where we exceed the target
index <- which(currentCoverage > targetCoverage)[1]

# Additonal number of people we need to vaccianted
target = vaccinesAvailable - (currentCoverage[index - 1] * wholePopulation)
# Actual additional people vaccinated in this step
actual = (currentCoverage[index] * wholePopulation) - (currentCoverage[index - 1] * wholePopulation)
# Ratio of target to actual
ratio = target / actual

# Identify the age group(s) that have increased in this step
stepped <- matrix[index,] > matrix[index - 1, ]

# Create the adjsuted matrix
adjustedMatrix <- matrix
adjustedMatrix[index,stepped] <- adjustedMatrix[index,stepped] * ratio

# Check the adusted matrix - perhaps these could be tests?
  # All rows except last are the same
all(adjustedMatrix[-nrow(adjustedMatrix), ] == matrix[-nrow(matrix), ])
  # Coverage of final row == targetCoverage
(sum(adjustedMatrix[nrow(adjustedMatrix),] * population) / wholePopulation) == targetCoverage
  # All columns a monotonically increasing
all(apply(adjustedMatrix, 2, function(x){
  all(diff(x) >= 0)
}))
```

Hopefully that makes sense?! It probably needs some checking on a few test cases once implemented

Reply:
(apologies if I've mis-read the JS and that feature is in there)

Reply:
Having thought about this overnight, I can imagine the above failing when more than 1 age groups is increased in the same step. I think this following adjustment solves this:
```

# matrix <- matrix(c(0, 0, 0.8, 0, 0.8, 0.8), nrow = 2, byrow = TRUE)
matrix <- matrix(c(0, 0, 0.8, 0, 0.8, 0.81), nrow = 2, byrow = TRUE)

population <- c(100, 100 , 100)
vaccinesAvailable <- 100

wholePopulation = sum(population)
targetCoverage = vaccinesAvailable / wholePopulation

# Estimate coverage for each step
currentCoverage = c()
for(i in 1:nrow(matrix)){
  currentCoverage[i] <- sum(matrix[i,] * population) / wholePopulation
}
# Identify where we exceed the target
index <- which(currentCoverage > targetCoverage)[1]

# Change in coverage
delta_cov <- matrix[index,] - matrix[index - 1,]
delta_cov_nor <- delta_cov / sum(delta_cov)

# Additional number of people we need to vaccinate
target = delta_cov_nor * (vaccinesAvailable - (currentCoverage[index - 1] * wholePopulation))

# Create the adjsuted matrix
adjustedMatrix <- matrix
adjustedMatrix[index,] <- adjustedMatrix[index - 1,] + (target / population)

# Check the adusted matrix - perhaps these could be tests?
# All rows except last are the same
all(adjustedMatrix[-nrow(adjustedMatrix), ] == matrix[-nrow(matrix), ])
# Coverage of final row == targetCoverage
(sum(adjustedMatrix[nrow(adjustedMatrix),] * population) / wholePopulation) == targetCoverage
# All columns a monotonically increasing
all(apply(adjustedMatrix, 2, function(x){
  all(diff(x) >= 0)
}))
```